IL1-SCIM.md

```diff
@@ -0,0 +1,149 @@
+
+
+# 3.0 Profile
+
+## 3.1 Client Authentication
+Identity Services shall authenticate to the Application's SCIM endpoint using OAuth 2.0 with the JWT Authorization Grant (RFC 7523). All of the following conditions should be met:
+* The token shall exchange a signed JWT for an access token and present that token in the {Authorization: Bearer} header on all subsequent SCIM requests.
+* The token must be scoped to "scim" and not grant broader permissions.
+* The token must contain a "token_endpoint" value which is the URL of the Application Provider's OAuth 2.0 token endpoint.
+
+Additionally, the Identity Service should expose a jwks_uri to allow the Application to perform signature verification
+
+## 3.2 SCIM Interoperability Requirements
+
+### 3.2.1 General Requirements
+* The Identity Service shall implement the required functionality of a SCIM Client as defined in RFC 7643 and RFC 7644.
+* The Application shall implement the required functionality of a SCIM Client as defined in RFC 7643 and RFC 7644.
+* All SCIM operations shall be authenticated via OAuth 2.0; local modifications are prohibited.
+
+### 3.2.2 User Provisioning Operations
+The Application MUST provide support all User provisioning operations defined in this section.
+
+#### 3.2.3 Create User (POST /Users)
+User creation is performed by the SCIM operation POST /Users
+
+#### 3.2.4 Update User (PATCH /Users/{id})
+User updates are performed via the SCIM operation: PATCH /Users/{id}
+
+#### 3.2.5 Deactivate or Reactivate User (PATCH /Users/{id})
+Changes to the user activation status are performed via the SCIM operation: PATCH /Users/{id}
+
+The Identity Service SHOULD propagate user deactivation events to the Application within 5 minutes of the user being deactivated.
+
+The Application SHOULD respond to user deactivation events by revoking the ability for the user to continue accessing the Application, including the revocation of currently active sessions. The revocation mechanism is an implementation detail outside the scope of this specification. Revocation SHOULD occur within 5 minutes of receiving the deactivation.
+
+The Application MUST allow reactivation of a deactivated user. 
+
+#### 3.2.6 Delete User (DELETE /Users/{id})
+User deletions are performed via the SCIM operation: DELETE /Users/{id}
+
+After a user is deleted, Application MUST allow the creation of a new user with the same username. 
+
+#### 3.2.7 Get User By ID (GET /Users/{id})
+User searches by id are performed via the SCIM operation: GET /Users/{id}
+
+#### 3.2.8 List Users By Alternate Identifier (GET /Users?)
+User searches by alternate identifier are performed via the SCIM operation: GET /Users?filter={filterExpression}
+
+Application Providers MUST support the following filter expressions:
+* username eq {username}
+* externalId eq {externalId}
+* emails[value eq {email}]
+
+### 3.3 Group (Role) Provisioning Operations
```

Comment:
Need to define audit operation to list all members of a group and how to handle large datasets (e.g page).  Identity Server needs to onboard brownfield app as well as ensure no drift between system of record and app

Reply:
Need to define how to remove a group (e.g https://openid.net/specs/fastfed-scim-1_0-02.html#SCIMInteroperabilityGroupProvisioningDeactivateGroup)